### PR TITLE
Check for standalone embedded objects

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

It is not allowed to create standalone object for tables marked as embedded. A check is added, and an exception is thrown.

This closes https://jira.mongodb.org/browse/RJS-636 (private ticket).

## ☑️ ToDos

* [x] 📝 Changelog entry
* [x] 📝 `Compatibility` label is updated or copied from previous entry
* [x] 🚦 Tests
* ~[ ] 📝 Public documentation PR created or is not necessary~
* ~[ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
* ~[ ] typescript definitions file is updated~
* ~[ ] jsdoc files updated~
* ~[ ] Chrome debug API is updated if API is available on React Native~

### CODE REVIEW

File: tests/js/list-tests.js [hunk 1]
```diff
@@ -1475,6 +1475,28 @@ module.exports = {
             TestCase.assertEqual(addresses[i]["street"], streets[i]);
         }
 
+        // creating standalone embedded object is not allowed
+        let addrs1 = owners[0].addresses;
```
Comment (blagoev) [inline tests/js/list-tests.js:1479]: shouldn't this be a separate test.

Comment (blagoev) [reply tests/js/list-tests.js:1479]: the logic from line 1478 onwards

File: tests/js/list-tests.js [hunk 2]
```diff
@@ -1475,6 +1475,28 @@ module.exports = {
...
+        realm.write(() => {
+            TestCase.assertThrows(() => {
+                realm.create(schemas.AddressSchema.name, { street: "Njalsgade", city: "Islands Brygge" });
+            });
+
+            TestCase.assertThrows(() => {
+                addrs1.push({ street: "Njalsgade", city: "Islands Brygge" });
```
Comment (steffenagger) [inline tests/js/list-tests.js:1486]: We're not allowed to push to an existing user's list of addresses either?

File: tests/js/list-tests.js (hunk 1 above)
Comment (kneth) [reply tests/js/list-tests.js:1479]: Sounds as a good idea

File: tests/js/list-tests.js (hunk 2 above)
Comment (kneth) [reply tests/js/list-tests.js:1486]: The above implies that you create an embedded object. Standalone or free-floating embedded objects are not allowed. Moreover, you cannot move an embedded object (removing it from it parent implies that it becomes free-floating).

Comment (kneth) [reply tests/js/list-tests.js:1486]: See "Semantics" in https://github.com/realm/realm-java/pull/6730

Comment (steffenagger) [reply tests/js/list-tests.js:1486]: But wouldn't this be equal to:
```java
managedObject.list.add(EmbeddedPerson()) // Will automatically copy the object as embedded
```
 (line from description @ https://github.com/realm/realm-java/pull/6730)

**Comment by kneth** [comment]
@blagoev @steffenagger I have changed it to a draft as I have discovered things we need cover before merging.

**Comment by blagoev** [comment]
LGTM for merge after the comments are resolved

File: tests/js/list-tests.js (hunk 2 above)
Comment (steffenagger) [reply tests/js/list-tests.js]: Fine as a temp crash fix.

Comment (kneth) [reply tests/js/list-tests.js]: We will reintroduce (fix might be a better word) the ability to push to a list later.

**Comment by kneth** [comment]
Tests pass locally (with and without sync)

**Pull request merged**